node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -14,23 +14,31 @@
 // You should have received a copy of the GNU General Public License
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
+mod executor_intf;
+mod memory_stats;
+
+pub use executor_intf::{prepare, prevalidate};
+
+// NOTE: Initializing logging in e.g. tests will not have an effect in the workers, as they are
+//       separate spawned processes. Run with e.g. `RUST_LOG=parachain::pvf-prepare-worker=trace`.
+const LOG_TARGET: &str = "parachain::pvf::prepare-worker";
```

Comment:
Is this really a legit `<target>::<subtarget>` log target ? I'd image we want `parachain::pvf-prepare-worker`

Reply:
Not sure... just had a look at the code and I think the target can just be an arbitrary string... I like the `::` syntax because it mimics the module separator in Rust, but I can change it.

https://docs.rs/tracing/latest/tracing/struct.Metadata.html#method.target

Reply:
I ran some tests:

```
$ RUST_LOG=parachain::pvf::execute-worker=trace cargo test terminates_on_timeout
[...]
May 25 10:48:25.074 DEBUG parachain::pvf::execute-worker: worker: validating artifact /var/folders/l5/3cfpm4g50_372y094h9g5ycc0000gp/T/.tmprmz4h4/wasmtime_0xee0ad047ee75d1a234e148f2acd1454719beda226a9315db6246172435ee0fd5_0x03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314 worker_pid=47156
May 25 10:48:28.130  WARN parachain::pvf::execute-worker: execute job took 3054ms cpu time, exceeded execute timeout 3000ms worker_pid=47156
```

```
$ RUST_LOG=parachain::pvf=trace cargo test terminates_on_timeout
[...]
May 25 10:48:38.782 DEBUG parachain::pvf::common: starting pvf worker (prepare) worker_pid=47177
May 25 10:48:38.785 DEBUG parachain::pvf::prepare-worker: worker: preparing artifact worker_pid=47177
May 25 10:48:38.795 DEBUG parachain::pvf::prepare-worker: worker: writing artifact to /var/folders/l5/3cfpm4g50_372y094h9g5ycc0000gp/T/.tmp5COv8L/prepare-artifact-RD95gcSfRz worker_pid=47177
May 25 10:48:38.806 DEBUG parachain::pvf::common: starting pvf worker (execute) worker_pid=47179
May 25 10:48:38.807 DEBUG parachain::pvf::execute-worker: worker: validating artifact /var/folders/l5/3cfpm4g50_372y094h9g5ycc0000gp/T/.tmp5COv8L/wasmtime_0xee0ad047ee75d1a234e148f2acd1454719beda226a9315db6246172435ee0fd5_0x03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314 worker_pid=47179
May 25 10:48:41.863  WARN parachain::pvf::execute-worker: execute job took 3053ms cpu time, exceeded execute timeout 3000ms worker_pid=47179
May 25 10:48:41.863 DEBUG parachain::pvf::common: quitting pvf worker (execute): Custom { kind: UnexpectedEof, error: "early eof" } worker_pid=47179
```

I'll document this in the `gum` crate, it currently it has no usage information.